### ISSUE TITLE
`ymap_destroy` doesn't exist

### ISSUE THREAD

**Issue by jakovpucekovic** [y-crdt/y-crdt#512]
Hi, 
while using the yffi lib I noticed that the comments of `ymap` says that `ymap_destroy` should be called to release the resources, but that function doesn't exist. Is the comment outdated/wrong? Should another function be called to release resources allocated when using the `ymap` function?

Sources mentioning `ymap_destroy`:
https://github.com/y-crdt/y-crdt/blob/74fb86e2b1a536c97859a9416d895643f5737fd4/tests-ffi/include/libyrs.h#L1295
https://github.com/y-crdt/y-crdt/blob/74fb86e2b1a536c97859a9416d895643f5737fd4/yffi/src/lib.rs#L747

Thanks in advance

**Comment by Horusiath** [comment]
Yes, the comment is outdated. You don't need to manually destroy shared types. If you delete them from their parent collection, they will be deallocated automatically and the pointer won't be valid anymore. 

If you need to refer to `Branch` pointers but don't know if they are valid or not, you can use `YBranchId ybranch_id(*Branch)` to generate logical identifier, and then `*Branch ybranch_get(*YBranchId, *Transaction)` to resolve the branch pointer back (which will be NULL pointer if the instance no longer exists or was not found).
